src/main/kotlin/lotto/service/LottoGame.kt

```diff
@@ -0,0 +1,33 @@
+package lotto.service
+
+import lotto.data.Lotto
+import lotto.data.LottoNumber
+import lotto.data.LottoRanking
+import lotto.data.WinningLotto
+import lotto.domain.LottoMachine
+import lotto.domain.RandomLogicInterface
+import lotto.domain.WinningDomain
+
+class LottoGame(private val randomLogic: RandomLogicInterface) {
+
+    fun buyLotto(cash: Int): List<Lotto> {
+        val lottoList = mutableListOf<Lotto>()
+        val times = cash / GAME_COST
+
+        repeat(times) {
+            val lottoNumberCombination = LottoNumber.createRandomLottoNumber(randomLogic)
+            lottoList.add(LottoMachine.createSelectLotto(lottoNumberCombination))
+        }
+
+        return lottoList
+    }
```

Comment:
이펙티브 시리즈 책의 내용중에는 `추상화의 레벨을 맞춰라` 라는 말이 있는데요. 
이는 다른 함수를 호출해서 결과를 만드는 것과 직접 로직을 작성하는게 혼용되지 말라는 의미를 가집니다. 

그리고 도메인에 책임을 위임하는게 아닌 직접 연산이 필요하거나 개념적 분리가 필요한 경우 함수로 분리를 하고는 하는데, 이 경우 이 분리의 수준을 맞추면 좋을 것 같아요. 

```suggestion
    fun buyLotto(cash: Int): List<Lotto> {
       val times = getTimes(cash)
       return createLotto(times)
    }
    
    fun getTimes(cash: Int) {
        return cash / GAME_COST // 여기도 전략패턴 반영 가능 하겠네요.
    }
    
    fun createLotto(times: Int) {
     // ...
    }
```

Reply:
로또 계산기의 역할이라 생각하여  getTimes를 LottoCalculator에서 구현하였습니다.
getTimes의 파라미터로 enum Country(naem: String, gameCost: Int)와 같은 구성의 클래스가 오면 자연스럽게 나라별 게임 횟수를 구할 수 있을 것 같습니다.